src/engine/renderer/gl_shader.h

```diff
@@ -589,7 +589,10 @@ class GLUniform4f : protected GLUniform
 #if defined( USE_UNIFORM_FIREWALL )
 		vec4_t *firewall = ( vec4_t * ) &p->uniformFirewall[ _firewallIndex ];
 
-		if ( Vector4Compare( *firewall, v ) )
+		if ( !( *firewall[ 0 ] != v[ 0 ] ||
```

Comment:
Moving this outside of q_shared.h might fasten a bit compilation, but I initially found it because of a warning about comparing floats. I think there is room for improvements, here.

Reply:
I'm not very happy with replacing reusable syntax with custom one as well…
What was the issue with floats?

Reply:
You could use `memcmp` here

Reply:
See that comment as I did some comparison on similar code before:

https://github.com/DaemonEngine/Daemon/pull/389#issuecomment-713729634

Reply:
Well, the point here is that the `Vector4Compare` inline is used only once in the whole daemon code base, and is in a header which is included from everywhere. So I don't think it's that useful. Maybe there are other functions doing the same thing around, though, and if yes, it would be interesting to use those instead of memcmp() (that I just pushed, this is a nice trick I'll remember it)